spacy/cli/package.py

```diff
@@ -173,6 +192,11 @@ def package(
             util.run_command([sys.executable, "setup.py", "bdist_wheel"], capture=False)
         wheel = main_path / "dist" / f"{model_name_v}{WHEEL_SUFFIX}"
         msg.good(f"Successfully created binary wheel", wheel)
+    if "__" in model_name:
+        msg.warn(
+            f"Model name ('{model_name}') contains a run of underscores. "
+            "These will be squashed to a single underscore when normalized upon PyPI upload.",
```

Comment:
We wouldn't recommend uploading model packages to pypi (they would normally be rejected due to size anyway), so this isn't the explanation. The reason is that pip doesn't treat it as a distinct package name.

```suggestion
            f"Model name ('{model_name}') contains a run of underscores. "
            "Since pip install squashes these to a single underscore, we recommend "
            "package names without sequences of underscores."
```

Reply:
I admit I wasn't completely understanding the implication here or the cause of the squashing. My understanding is now as follows:

`setuptools`, via `python setup.py build_wheel`, (not `pip`) will squash the run of underscores when the wheel is created. 

e.g.
```
python -m spacy package model-best output/ --name double__underscore --build wheel
```

Results in a wheel file at `output/en_double__underscore-0.0.0/dist/en_double_underscore-0.0.0-py3-none-any.whl` 
 
Which means we would need to fix the success message, because it's using the unsquashed name (this file does not exist):
```
 ✔ Successfully created binary wheel
output/en_double__underscore-0.0.0/dist/en_double__underscore-0.0.0-py3-none-any.whl
```
 
`pip` gets implicated now when a user chooses to install the package if they're pulling it from a repository, because "Comparison of project names is case insensitive and treats arbitrarily-long runs of underscores, hyphens, and/or periods as equal." Meaning, this model package from a repository could be installed with `pip install double__underscore` or `pip install double_underscore`. If there's already a `double_underscore` or `double-underscore` package on that repository, it will result in a conflict when they attempt to upload.

I haven't fully explored the world of python packaging so some of my understanding could still be wrong here. It also does this for the `egg-info` directory that's created, but I'm not sure if that matters.

---

All that said, I think our solution should remain mostly the same:
- We should warn them that `pip` (on install) and `setuptools` (on wheel creation) squash runs of underscores, and link to https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/#name
- We should correct the success message for wheel creation to point to the correct file

Reply:
I don't think users typically care much whether it's `pip` or `setuptools`, and the project names also still don't have the same restrictions we're focusing on here, so I wouldn't include that link (all the examples with `-` and `.` are confusing) and I would focus on:

* warn that runs of underscores are not significant in installed package names
* fix the reported wheel filename (yeesh on the sdist vs. wheel names here)

Reply:
Sounds good! I was trying to anticipate where a user would encounter an issue if they ignored the warning, but that might be too clever/detailed. I updated with the general warning you proposed.